### PR TITLE
DPL: speedup forwarding in case of split payload parts

### PR DESCRIPTION
**Pull request by ktf**

### CODE REVIEW

**Comment by ktf** [comment]
@davidrohr @shahor02 @chiarazampolli this should speedup the dispatcher.

File: Framework/Core/src/DataProcessingDevice.cxx [hunk 1]
```diff
@@ -1205,45 +1205,60 @@ bool DataProcessingDevice::tryDispatchComputation(DataProcessorContext& context,
...
+        // We need to find the forward route only for the first
+        // part of a split payload. All the others will use the same.
+        if (fdh->splitPayloadIndex == 0) {
+          cachedForwardingChoice = -1;
+          for (size_t fi = 0; fi < spec->forwards.size(); fi++) {
+            auto& forward = spec->forwards[fi];
+            if (DataSpecUtils::match(forward.matcher, dh->dataOrigin, dh->dataDescription, dh->subSpecification) == false || (dph->startTime % forward.maxTimeslices) != forward.timeslice) {
+              continue;
+            }
+            cachedForwardingChoice = fi;
```
Comment (shahor02) [inline Framework/Core/src/DataProcessingDevice.cxx:1241]: should not one break here?

**Comment by davidrohr** [comment]
This works, and #7235 also works.
#7235 has 30% less CPU load in the dispatcher compared to this one.

File: Framework/Core/src/DataProcessingDevice.cxx (hunk 1 above)
Comment (ktf) [reply Framework/Core/src/DataProcessingDevice.cxx:1241]:
```suggestion
            cachedForwardingChoice = fi;
            break;
```

Comment (ktf) [reply Framework/Core/src/DataProcessingDevice.cxx:1241]: yes.

**Review by davidrohr — APPROVED** [review]

**Comment by ktf** [comment]
Tested by @davidrohr and myself. Seems quite an improvement, so we merge it to get it in tomorrow build.

**Pull request merged**